src/groupeddataframe/grouping.jl

```diff
@@ -223,9 +230,9 @@ function Base.map(f::Any, gd::GroupedDataFrame)
         resize!(starts, j-1)
         resize!(ends, j-1)
         ends[end] = length(idx)
-        return GroupedDataFrame(parent, gd.cols, collect(1:length(idx)), starts, ends)
+        return GroupedDataFrame(parent, gd.cols, idx, collect(1:length(idx)), starts, ends)
     else
-        return GroupedDataFrame(parent, gd.cols, Int[], Int[], Int[])
+        return GroupedDataFrame(parent, gd.cols, idx, Int[], Int[], Int[])
```

Comment:
`idx` is undefined in this branch. Can you please also add a test that catches it?